coverage-tool/coverage-tool.py

```diff
@@ -0,0 +1,300 @@
+
+        # If property is an enum, find coverage for each enum value
+        if "enum" in prop_content:
+            for enum in prop_content["enum"]:
+                coverage_files = []
+                for file_name in responses:
+                    if any(
+                        enum == part.get(prop_name) for part in responses[file_name]
+                    ):
+                        coverage_files.append(file_name)
+                total_fields += 1
+                if coverage_files:
+                    covered_fields += 1
+                output[prop_name][enum] = {NUM_KEYWORD: (len(coverage_files))}
+                if args.list_files:
+                    output[prop_name][enum][FILES_KEYWORD] = coverage_files
+
+        # If property is an instance of a schema, find coverage for that schema
+        if "$ref" in prop_content:
+            # Move into the referenced property in each response
+            next_responses = {
+                f: [part[prop_name] for part in responses[f] if prop_name in part]
+                for f in responses
+            }
+            # Recursively find coverage of referenced schema with updated responses
+            output[prop_name].update(
+                find_coverage(
+                    schemas[prop_content["$ref"]]["properties"], next_responses
+                )
+            )
+
+        # If property is an array of a schema, find coverage for each instance
+        if prop_content.get("type") == "array" and "$ref" in prop_content["items"]:
+            next_responses = {
+                f: [elem for part in responses[f] for elem in part.get(prop_name, [])]
+                for f in responses
+            }
+            output[prop_name].update(
+                find_coverage(
+                    schemas[prop_content["items"]["$ref"]]["properties"], next_responses
+                )
+            )
+    return output
+
+
+def print_output(output, indent=0, red_indent=0):
+    """Print the coverage data."""
+    if args.percent_only:
+        print(output[TOTAL_KEYWORD])
+    elif args.json_output:
+        print(json.dumps(output, indent=2))
+    else:
+        color_red, color_end = "\033[91m", "\033[0m"
+        for key, value in output.items():
+            if key == TOTAL_KEYWORD:
+                print(f"Total Coverage: {value[NUM_KEYWORD]}%")
+            elif key not in {NUM_KEYWORD, FILES_KEYWORD}:
+                color = not args.no_color and value[NUM_KEYWORD] == 0
+                print(
+                    "| " * (indent - red_indent)
+                    + (color_red if color else "")
+                    + "| " * red_indent
+                    + f"{key}: {value[NUM_KEYWORD]} "
+                    + str(value.get(FILES_KEYWORD) or "")
+                    + (color_end if color else "")
+                )
+                print_output(value, indent + 1, red_indent + (color))
+
+
+args = get_args()
+schemas = get_schemas()
```

Comment:
If I run the tool once locally I'll never get an updated schema, right?

Reply:
We're fetching the schema on every run.